cmd/step/main.go

```diff
@@ -128,49 +129,78 @@ func Run(ctx context.Context, step Step) error {
 	}
 
 	for i, command := range step.Commands {
-		//nolint:gosec
-		cmd := exec.CommandContext(ctx, step.Shell, "-c", command)
-		cmd.Env = os.Environ()
+		attempt := 0
+		var lastErr error
+
+		for attempt <= step.RetryAttempts {
+			if attempt > 0 {
+				log.WithFields(log.Fields{
+					"attempt": attempt,
+					"command": i,
```

Comment:
do you want the command or the index of it here and below?

Reply:
as discussed - made it index